### PR TITLE
Introduce an edit field to Date Picker in quick filters and leases

### PR DESCRIPTION
**Pull request by abdelrahmansd**
## What does this change?
this makes the Datepicker's edit field visible so it allows the user to easily enter the dates

## How can success be measured?
User can type the date in the Datepicker's text input and the calendar should sync with the text input

## Screenshots

Screenshot from 2021-03-26 03-50-21

## Tested?
- [x] locally by committer
- [ ] locally by Guardian reviewer
- [ ] on the Guardian's TEST environment

### CODE REVIEW

**Comment by abdelrahmansd** [comment]
Hi @AWare Thanks for the review, I think this will be different than the [proposal](https://github.com/guardian/grid/issues/3221),
@akash1810 what do you think?

**Comment by akash1810** [comment]
@AWare IIUC those links suggest making dates more accessible by making the input more human readable? Sounds like a great idea to me and would suggest that as a different change. I don't think it blocks this PR.

On a related note, Grid already has some [limited support for relative dates](https://github.com/guardian/grid/blob/4813284f701cede49b4465bacc86817877c3a43e/media-api/test/scala/lib/querysyntax/ParserTest.scala#L195-L235), joining this new field to that query string might be possible.

The changes here look good, I'll put them on our pre-PROD environment to test them and will merge once I have.

Thanks for the contribution and apologies my late review!

File: kahuna/public/js/components/gu-date-range/gu-date-range.html
Comment (akash1810) [inline kahuna/public/js/components/gu-date-range/gu-date-range.html:59]: Is this meant to be hidden?

File: kahuna/public/js/components/gu-date-range/gu-date-range.html
Comment (akash1810) [inline kahuna/public/js/components/gu-date-range/gu-date-range.html:70]: Is this meant to be hidden?

File: kahuna/public/js/components/gu-date/gu-date.html
Comment (akash1810) [inline kahuna/public/js/components/gu-date/gu-date.html:20]: Should we add a placeholder here?

**Review by akash1810 — COMMENTED** [review]
Have deployed this to our pre-PROD environment now. The screenshot of the change doesn't match what we're seeing.

We're seeing this on the `gu-date-range` component (no date input):
image

This is down to the `hidden` attribute.

For the `gu-date` we're seeing this:
image

I think it would be preferable for the input field to be shown in the date modal, rather than at the top level of the component.

**Comment by paperboyo** [comment]
@abdelrahmansd This looks like it could support a user request that we had here: for Leases to allow choosing hours (not only days). Similarly, it could be useful for setting ranges for search? Of course, only if that is indeed possible, which I judge *purely* by seeing this user-hostile looking UTC datetime string (we would need to have it formatted in a nice way).

Disregard this comment if this isn’t easy!

File: kahuna/public/js/components/gu-date-range/gu-date-range.html
Comment (abdelrahmansd) [reply kahuna/public/js/components/gu-date-range/gu-date-range.html]: fixed

File: kahuna/public/js/components/gu-date-range/gu-date-range.html
Comment (abdelrahmansd) [reply kahuna/public/js/components/gu-date-range/gu-date-range.html]: fixed

File: kahuna/public/js/components/gu-date/gu-date.js
Comment (akash1810) [inline kahuna/public/js/components/gu-date/gu-date.js:48]: Can I suggest using `DISPLAY_FORMAT` defined above on L10? The `DD-MM-YYYY` format can be a little ambiguous for our American friends - is 03/04/2021 03 April or 04 March?

`DISPLAY_FORMAT` is `DD MMM YYYY` which will render 03/04/2021 as 03 Mar 2021, which removes the ambiguity.

**Review by akash1810 — COMMENTED** [review]
Looks great! A nitpick comment on the date format, which applies to each field.

File: kahuna/public/js/components/gu-date/gu-date.js
Comment (abdelrahmansd) [reply kahuna/public/js/components/gu-date/gu-date.js:48]: yes, make sense

**Review by akash1810 — APPROVED** [review]
Have tested these changes on our pre-production environment and they look good to me 👍🏽 .

Will merge in the morning.

Apologies for the delay and thanks for your patience!

**Pull request merged**
